### ISSUE TITLE
[PM-1123] Pegasus is giving error in copying/transferring files.

### ISSUE THREAD

**Issue by mayani** [pegasus-isi/pegasus#1237]
I an new to Pegasus so do not known exactly how the files are transferred. When I try to launch a 'PyCBC' pipeline I get stuck with the following. 

1. Attempt is made to create a soft link of existing file in 'CONDOR EXECUTE'
  and no neither link is created nor error is reported.
2. Attempt is made to copy file from  'CONDOR EXECUTE' to some directly and error is reported that the file does not exist. 

I do not know what I am missing. Please have a look at the following error message. 

###################### staging in input data and executables ######################
2016-09-15 11:29:39,922    INFO:  Reading URL pairs from stdin
2016-09-15 11:29:39,922    INFO:  1 transfers loaded
2016-09-15 11:29:39,922    INFO:  PATH=/var/lib/condor/execute/dir_659992/pegasus.IS9kPt/pegasus-4.6.2/bin:/home/jayanti/opt/pycbc/bin:/home/jayanti/opt/pylal/bin:/soft/lscsoft/glue/bin:/soft/lscsoft/lalsuite/bin:/usr/lib64/qt-3.3/bin:/home/jayanti/perl5/bin:/usr/local/bin:/bin:/usr/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/jayanti/.local/bin:/home/jayanti/bin
2016-09-15 11:29:39,923    INFO:  LD_LIBRARY_PATH=/home/jayanti/opt/pylal/lib64/python2.7/site-packages:/soft/lscsoft/glue/lib64/python2.7/site-packages:/soft/gsl/lib:/soft/fftw/lib:/soft/lscsoft/libframe/lib/:/soft/lscsoft/lalsuite/lib:/soft/metaio/lib/:/soft/hdf5-1.8.12/lib:/soft/fftw/lib/:/soft/healpix/lib/:/usr/lib
2016-09-15 11:29:39,961    INFO:  --------------------------------------------------------------------------------
2016-09-15 11:29:39,961    INFO:  Starting transfers - attempt 1
2016-09-15 11:29:41,965    INFO:  ln -f -s '/home/jayanti/LIGO/pycbc/make_psd/test1/output/local-site-scratch/work/./main_ID0000001/H1L1-WORKFLOW_SEGMENT_SUMMARY-1123856384-2760704.xml' '/var/lib/condor/execute/dir_659992/pegasus.IS9kPt/H1L1-WORKFLOW_SEGMENT_SUMMARY-1123856384-2760704.xml'
2016-09-15 11:29:41,974    INFO:  --------------------------------------------------------------------------------
2016-09-15 11:29:41,974    INFO:  Stats: Total 1 transfers, 69.3 KB transferred in 2 seconds. Rate: 33.8 KB/s (270.1 Kb/s)
2016-09-15 11:29:41,974    INFO:         Between sites local->local : 1 transfers, 69.3 KB transferred in 2 seconds. Rate: 33.8 KB/s (270.1 Kb/s)
2016-09-15 11:29:41,975    INFO:  All transfers completed successfully.

############################# staging out output files #############################
2016-09-15 11:29:42,593    INFO:  Reading URL pairs from stdin
2016-09-15 11:29:42,594    INFO:  1 transfers loaded
2016-09-15 11:29:42,594    INFO:  PATH=/var/lib/condor/execute/dir_659992/pegasus.IS9kPt/pegasus-4.6.2/bin:/home/jayanti/opt/pycbc/bin:/home/jayanti/opt/pylal/bin:/soft/lscsoft/glue/bin:/soft/lscsoft/lalsuite/bin:/usr/lib64/qt-3.3/bin:/home/jayanti/perl5/bin:/usr/local/bin:/bin:/usr/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/jayanti/.local/bin:/home/jayanti/bin
2016-09-15 11:29:42,594    INFO:  LD_LIBRARY_PATH=/home/jayanti/opt/pylal/lib64/python2.7/site-packages:/soft/lscsoft/glue/lib64/python2.7/site-packages:/soft/gsl/lib:/soft/fftw/lib:/soft/lscsoft/libframe/lib/:/soft/lscsoft/lalsuite/lib:/soft/metaio/lib/:/soft/hdf5-1.8.12/lib:/soft/fftw/lib/:/soft/healpix/lib/:/usr/lib
2016-09-15 11:29:42,634    INFO:  --------------------------------------------------------------------------------
2016-09-15 11:29:42,634    INFO:  Starting transfers - attempt 1
2016-09-15 11:29:44,637    INFO:  /bin/cp -f -R -L '/var/lib/condor/execute/dir_659992/pegasus.IS9kPt/H1L1-PAGE_SEGTABLE_SUMMARY-1123856384-2760704.html' '/home/jayanti/LIGO/pycbc/make_psd/test1/output/local-site-scratch/work/./main_ID0000001/H1L1-PAGE_SEGTABLE_SUMMARY-1123856384-2760704.html'
2016-09-15 11:29:44,646    INFO:  /bin/cp: cannot stat ‘/var/lib/condor/execute/dir_659992/pegasus.IS9kPt/H1L1-PAGE_SEGTABLE_SUMMARY-1123856384-2760704.html’: No such file or directory
2016-09-15 11:29:44,646   ERROR:  Command exited with non-zero exit code (1): /bin/cp ...
2016-09-15 11:30:11,674    INFO:  --------------------------------------------------------------------------------

**Reporter:** @jayanti-prasad
**Resolution:** Fixed
**Watchers:**
@duncan-brown
@jayanti-prasad
@vahi

**Comment by mayani** [comment]
**Author:** @vahi

‘/var/lib/condor/execute/dir_659992/pegasus.IS9kPt/H1L1-PAGE_SEGTABLE_SUMMARY-1123856384-2760704.html’: No such file or directory 

Looks like the ligo job code did not run. 
you can run pegasus-analyzer . Or you can look at the .out.000 file for the job. That will have more information.

**Comment by mayani** [comment]
**Author:** @jayanti-prasad

(1) I know that there may be some issue in my condor configuration setting. In particular setting of 'EXECUTE'.

(2) My error file (plot_segments-SCIENCE_MINUS_CAT1-H1L1_ID13_ID0000013.err.000) says :

2016-09-21 10:58:44: PegasusLite: version 4.6.2
################################ Setting up workdir ################################
2016-09-21 10:58:44: Checking /var/lib/condor/execute/dir_2491918 for potential use as work space...
2016-09-21 10:58:44:   Work dir is /var/lib/condor/execute/dir_2491918/pegasus.BeNsrx - 37765064 kB available

the last line is not correct since there is no such file/dir on the submit node.

(3)Another section:
###################### staging in input data and executables ######################
2016-09-21 10:58:44,552    INFO:  Reading URL pairs from stdin
2016-09-21 10:58:44,553    INFO:  1 transfers loaded
2016-09-21 10:58:44,553    INFO:  PATH=/var/lib/condor/execute/dir_2491918/pegasus.BeNsrx/pegasus-4.6.2/bin:/home/jayanti/opt/pycbc/bin:/home/jayanti/opt/pylal/bin:/soft/lscsoft/glue/bin:/soft/lscsoft/lalsuite/bin:/usr/lib64/qt-3.3/bin:/home/jayanti/perl5/bin:/usr/local/bin:/bin:/usr/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/soft/healpix/bin:/home/jayanti/.local/bin:/home/jayanti/bin
2016-09-21 10:58:44,553    INFO:  LD_LIBRARY_PATH=/home/jayanti/opt/pylal/lib64/python2.7/site-packages:/soft/lscsoft/glue/lib64/python2.7/site-packages:/soft/gsl/lib:/soft/fftw/lib:/soft/cfitsio/lib/:/soft/gsl/lib/:/soft/fftw/lib/:/soft/hdf5/lib/:/soft/healpix/lib/:/soft/pgplot/:/soft/lscsoft/libframe/lib/:/soft/lscsoft/lalsuite/lib:/soft/metaio/lib/:/soft/hdf5-1.8.12/lib:/soft/fftw/lib/:/soft/healpix/lib/:/usr/lib
2016-09-21 10:58:44,598    INFO:  --------------------------------------------------------------------------------
2016-09-21 10:58:44,598    INFO:  Starting transfers - attempt 1
2016-09-21 10:58:46,602    INFO:  ln -f -s '/home/jayanti/LIGO/pycbc/make_psd/test1/output/local-site-scratch/work/./main_ID0000001/H1L1-SCIENCE-1123856384-2760704.xml' '/var/lib/condor/execute/dir_2491918/pegasus.BeNsrx/H1L1-SCIENCE-1123856384-2760704.xml'
2016-09-21 10:58:46,611    INFO:  --------------------------------------------------------------------------------
2016-09-21 10:58:46,612    INFO:  Stats: Total 1 transfers, 26.4 KB transferred in 2 seconds. Rate: 12.8 KB/s (102.5 Kb/s)
2016-09-21 10:58:46,612    INFO:         Between sites local->local : 1 transfers, 26.4 KB transferred in 2 seconds. Rate: 12.8 KB/s (102.5 Kb/s)
2016-09-21 10:58:46,612    INFO:  All transfers completed successfully.

again this is not correct. No symlink as is tried was created and so again the message of the last line is nor correct.

(3)
############################# staging out output files #############################
2016-09-21 10:58:49,228    INFO:  /bin/cp -f -R -L '/var/lib/condor/execute/dir_2491918/pegasus.BeNsrx/H1L1-PLOT_SEGMENTS_SCIENCE_MINUS_CAT1-1123856384-2760704.html' '/home/jayanti/LIGO/pycbc/make_psd/test1/output/local-site-scratch/work/./main_ID0000001/H1L1-PLOT_SEGMENTS_SCIENCE_MINUS_CAT1-1123856384-2760704.html'
2016-09-21 10:58:49,237    INFO:  /bin/cp: cannot stat ‘/var/lib/condor/execute/dir_2491918/pegasus.BeNsrx/H1L1-PLOT_SEGMENTS_SCIENCE_MINUS_CAT1-1123856384-2760704.html’: No such file or directory

this needs no explanation. since (1) and (2) have already failed. 

I do not know what to check. We have a NFS system so I am wondering if there is any easy way to transfer files to compute nodes. 

thanks.

**Comment by mayani** [comment]
**Author:** @jayanti-prasad

I was able to run the pipeline successfully without making any change in my condor set up. The problem was with my PyCBC installation and ini file I was using. 

I think  the error messages (as I posted above ) from *.err.* files are misleading.

For some jobs (for which executables and libraries are fine) transfer happens without any issue.

Jobs which have issues give errors like:

INFO:  /bin/cp: cannot stat ‘/var/lib/condor/execute/dir_3651855/pegasus.CiW5XP/H1L1-PLOT_RANGE_BBH1-1126051217-1220400.png’: No such file or directory

It would be great if the error message clearly says why the jobs failed (executables is not in path, libraries not available,
arguments are not correct etc.,)
